### DISCUSSION TITLE
Sana Bayılıyorum 1543

### DISCUSSION THREAD

**Discussion by PourToujoursMysterieux** [sonsuzus/sonsuzus#467] [Soru Cevap]
Bir gün Polycarp uyandığında en sevdiği sayıının 1543 olduğunu fark etti.

O gün uyandığında ilk gördüğü boyutları n'ye m olan uzun bir duvar halısı gördü.M ve N tam sayılardır.Her hücre 0 ile 9 arasındadır(o ve 9'a eşit olabilir.)

Polycarp 1543 sayısının saat yönünün dışına doğru gidildiğinde her katmanda* kaç tane 1543 oluştuğunu merak etti.
* Her katmanın boyutu yani n.m 2.(n+m - 2) olarak tanmlanmıştır.Ayrıca her elementin kalınlığı,dış kısmını kaplar.Her sonraki katman halının birinci katmanı olarak tanımlanır ve önceki katmanları orjinal halıdan çıkarılarak elde edilir.
*
Girdi
Girdinin ilk satırı bir integer içerir t(1 <= t <= 100)-test sayılarının sayısı.
Sıradaki satırlar test sayılarını açıklar.

Her satırdaki test sayısı  birtakım sayı içerir n ve m(2 <= n,m <= 10³, n,m).

Bu m uzunluğunda n tane satırdan oluşan, 0 ve 9 arası sayılardan oluşan halının tanımıyla devam eder.

n*m'nin 10⁶'yı aşmayacağı kesindir.

Çıktı
Çıktıda her test sayısı  için tek bir sayı  bulunur. Bu sayı, halıda saat yönünde dışa doğru 1543'ün belirdiği seferlerin toplamıdır. 

örnek girdi ve çıktılar:
girdi:
8
2 4
1543
7777
2 4
7154
8903
2 4
3451
8888
2 2
54
13
2 2
51
43
2 6
432015
512034
4 4
5431
1435
5518
7634
6 4
5432
1152
4542
2432
2302
5942

çıktı:
1 
1 
0 
1 
0 
2 
2 
2

**Comment by PourToujoursMysterieux** [comment]
```
def count_1543_in_layers (n,m):
    if not (0 <= n <= 9 and 0 <= m <= 9):
         return "n ve m değerleri 0 ile 9 arasında olmalıdır."

         total_1543_count = 0
         layer_count = min(n,m) // 2
         for _ in range(layer_count + 1):
             total_1543_count += (n*m + 2 * (n + m-2)) // 4
             n, m = n - 2, m- 2
         return total_1543_count


         n, m = 9 , 7
         print(count_1543_in_layers(n,m))

```

**Reply by sonsuzus** [reply]
Kod çalışmıyor ayrıca yapay zeka ürünü gibi tamamen. Test edilmeden konulmuş.
